src/object/srv_obj.c

```diff
@@ -1696,8 +1696,12 @@ do_obj_ioc_begin(uint32_t rpc_map_ver, uuid_t pool_uuid,
 		if (opc == DAOS_OBJ_RPC_CPD)
 			D_GOTO(out, rc = -DER_TX_RESTART);
 
-		if (obj_is_modification_opc(opc))
-			D_GOTO(out, rc = -DER_STALE);
+		/* NB: After Online server addition, all shards might be
+		 * be repositioned, so let's return -DER_STALE to force
+		 * client to refresh the pool map and retry for all
+		 * operations.
+		 */
+		D_GOTO(out, rc = -DER_STALE);
 		/* It is harmless if fetch with old pool map version. */
```

Comment:
This comment needs be removed, and actually I don't see why fetching with old pool map is harmless (even without OSA involved)

Reply:
Please remove this incorrect comment when the patch needs be refreshed.